### ISSUE TITLE
IPFS PubSub for Dapp

### ISSUE THREAD

**Issue by SionoiS** [brave/brave-browser#13607]
## Description 
Please consider enabling pubsub for IPFS in Brave. Pubsub would allow web apps to be a lot more dynamic. Static content is a good start but go-ipfs can do a lot more! IPNS and any kind of decentralized database will need it at some point too.

Security is a concern I'm not sure how to address though.

Thank you!

**Comment by dxtr85** [comment]
I second that.
It would be great having access to ipfs instance (at least when ipfs-companion is running with embedded node) from within every page for js scripts to interact directly with ipfs network - this would enable making dApps truly decentralized.

**Comment by SionoiS** [comment]
Yes, requiring the user to configure IPFS for CORS is less than ideal, that's another issue.

**Comment by slavnycoder** [comment]
yep! 

- [ ] API.HTTPHeaders.Access-Control-Allow-Origin '["*"]' by default
- [ ] pubsub by default

**Comment by daviddahl** [comment]
I also need this feature. Asking app users to install IPFS and configure it is a non-starter. Brave with pubsub and IPFS API will be transformative for decentralization on the web.
